src/client/index.tsx

```diff
@@ -16,53 +16,56 @@ __webpack_public_path__ = getPublicPath();
 import { nbformat } from '@jupyterlab/coreutils';
 import * as React from 'react';
 import * as ReactDOM from 'react-dom';
+import { NotebookOutputEventParams } from 'vscode-notebook-renderer';
+import { JupyterNotebookRenderer } from './constants';
 import { CellOutput } from './render';
 
+const notebookApi = acquireNotebookRendererApi(JupyterNotebookRenderer);
+
+notebookApi.onDidCreateOutput(renderOutput);
+
 /**
  * Called from renderer to render output.
  * This will be exposed as a public method on window for renderer to render output.
  */
-function renderOutput(tag: HTMLScriptElement) {
-    let container: HTMLElement;
-    const mimeType = tag.dataset.mimeType as string;
+function renderOutput(request: NotebookOutputEventParams) {
     try {
-        const output = JSON.parse(tag.innerHTML) as nbformat.IExecuteResult | nbformat.IDisplayData;
-        console.log(`Rendering mimeType ${mimeType}`);
-
-        // Create an element to render in, or reuse a previous element.
-        if (tag.nextElementSibling instanceof HTMLDivElement) {
-            container = tag.nextElementSibling;
-            container.innerHTML = '';
-        } else {
-            container = document.createElement('div');
-            tag.parentNode?.insertBefore(container, tag.nextSibling); // NOSONAR
-        }
-        tag.parentElement?.removeChild(tag); // NOSONAR
+        console.error('request', request);
+        const output = convertVSCodeOutputToExecutResultOrDisplayData(request);
+        console.log(`Rendering mimeType ${request.mimeType}`, output);
+        console.error('request output', output);
 
-        ReactDOM.render(React.createElement(CellOutput, { mimeType, output }, null), container);
+        ReactDOM.render(React.createElement(CellOutput, { mimeType: request.mimeType, output }, null), request.element);
```

Comment:
Oh no cellOutput is still used. We need the changes made to our rendering code ported as well.

Reply:
This file here probably needs some changes to get escaping to work correctly. I'll add an issue

Reply:
https://github.com/microsoft/vscode-notebook-renderers/blob/master/src/client/render.tsx

Reply:
I'll create a seprate issue for that.

Reply:
@rchiodo I have added an issue for this & made a reference to the original PR you submitted.